docs/src/explanations/Documentation-of-functions.md

```diff
@@ -0,0 +1,341 @@
+---
+
+layout: docs
+
+title:  "Documention of functions"
+
+section: "chisel3"
+
+---
+
+# Documentation of Functions := vs <>
+
+The difference in the function of := and <> has been a major source of debate in lots of conversations
+
+This document explains the difference by setting up an experiment using Scastie examples + DecoupledIO examples to put the functions to the test.
+
+
+
+### Experiment
+
+```scala mdoc
+// Imports used by the following examples
+import chisel3._
+import chisel3.util.DecoupledIO
+```
+
+The diagram for the experiment can be viewed **[here](https://docs.google.com/document/d/14C918Hdahk2xOGSJJBT-ZVqAx99_hg3JQIq-vaaifQU/edit?usp=sharing)**
+
+
+```scala mdoc:silent
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  // connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc
+ChiselStage.emitVerilog(new Wrapper)
+```
+## Concept 1: <> is Communicative
+
+
+
+This experiment is set up to test for the function of `<>` using the experiment above.
+
+Achieving this involves flipping the RHS and LHS of each arrow and seeing how '<>'  will react.
+( Scastie link for the experiment:https://scastie.scala-lang.org/Shorla/LVhlbkFQQnq7X3trHfgZZQ )
+
+
+
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect Producer to IO
+  io.in <> p.io.a
+  // connect producer to consumer
+  p.io.b <> c.io.a
+  // connect consumer to IO
+  c.io.b <> io.out
+}
+
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala modc
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion: 
+The Verilog remained the same without incurring errors, showing that the `<>` operator is generally communicative.
+
+
+
+
+## Concept 2: := means assign ALL signals from the RHS to the LHS, regardless of their direction.
+Using the same experiment code as above, we set to test for the function of ":="
+To achieve this, "<>" is being replaced with ":=" in the sample code above and the RHS and LHS are flipped.
```

Comment:
```suggestion
To achieve this, "<>" is being replaced with ":=" in the sample code above and the RHS and LHS are flipped.
```

Why are we also doing the RHS and LHS flipping in this experiment? I think it is conflating two different concepts. Shouldn't we just do an experiment where we start from the original and replace `<>` with `:=`?

Reply:
I corrected that now